### PR TITLE
chore: Upgrade to Yarn 4

### PR DESCRIPTION
**Pull request by Saadnajmi**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

https://yarnpkg.com/blog/release/4.0

The update seems safe for us, let's see what happens when we update CI. 

### Verification

CI should pass

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by samuelfreiberg** [comment]
I'm guessing the E2E tests are failing because the drivers aren't being hoisted to the E2E package.

This was previously done in the E2E package.json. Perhaps this was deprecated and there's a new way to do it?

  "installConfig": {
    "hoistingLimits": "dependencies"
  },

**Comment by Saadnajmi** [comment]
> I'm guessing the E2E tests are failing because the drivers aren't being hoisted to the E2E package.
> 
> This was previously done in the E2E package.json. Perhaps this was deprecated and there's a new way to do it?
> 
> "installConfig": { "hoistingLimits": "dependencies" },

Oh, thanks! I'll take a look. I didn't see anything to that effect int he change notes

**Comment by Saadnajmi** [comment]
@samuelfreiberg No, it seems installconfig.hostingLimits didn't change: https://yarnpkg.com/configuration/manifest#installConfig.hoistingLimits

**Comment by acoates-ms** [comment]
I had issues moving to node 18 with E2E apps. In the RN 73 PR I had to bump to node 18, which also involved bumping all the wdio / winappdriver packages.
We should probably hold off on this PR until that one merges at this point.

**Comment by Saadnajmi** [comment]
> I had issues moving to node 18 with E2E apps. In the RN 73 PR I had to bump to node 18, which also involved bumping all the wdio / winappdriver packages. We should probably hold off on this PR until that one merges at this point.

Yeah, that was the plan to reduce diffs. I'm hoping your Node 18 bump does it better than mine, and that was the actual thing preventing upgrading here.

**Comment by rurikoaraki** [comment]
We probably need to update the other yarn.lock in tester_deps too?

**Comment by Saadnajmi** [comment]
@rurikoaraki @acoates-ms this should be ready now

File: .yarnrc.yml
Comment (Saadnajmi) [reply .yarnrc.yml]: Right. Yarn kept adding this, I'll remove and see if it comes back

**Review by tido64 — APPROVED** [review]

**Review by rurikoaraki — APPROVED** [review]

**Pull request merged**
